apps/dashboard/app/models/posix_file.rb

```diff
@@ -91,6 +91,27 @@ def editable?
     path.file? && path.readable? && path.writable?
   end
 
+  def touch
+    FileUtils.touch(path)
+  end
+
+  def mv_from(src)
+    FileUtils.mv(src, path)
+  end
+
+  def handle_upload(tempfile)
+    path.parent.mkpath unless path.parent.directory?
+
+    FileUtils.mv tempfile, path.to_s
+
+    # umasks apply on top of 666 (-rw-rw-rw-) permissions. So a u mask of 022 would result
+    # in 666 - 022 = 644. Umasks are base 8 octal (what all this stuff expects).
+    mode = 0666 & (0777 ^ File.umask)
```

Comment:
Yes, it was Robin who fixed it originally.

It depends on how technically correct the comments should be. Bitmasks aren't technically subtracting anything, but the explanation in the comment is conveniently giving the correct answer anyhow in this case. If the umask was `0011` (for some reason) in the example, the answer would still be `0666` of course, and not `0655` like the perceived subtraction logic might suggest.

Is it necessary to explain how a bitmask works in the comments or could it just say:
```
# Apply the user's umask on top of 0666 (-rw-rw-rw-), since the file doesn't need to be executable.
```

Reply:
Yea I think it's just the subtraction operator that's throwing me off because that was in fact the bug. It may only be me because I wrote both the comment and the bug.

I think that comment you've given is appropriate. bitwise operators are hard enough, then throw on the erroneous comment about subtraction I think is enough to confuse anyone.